include/vcpkg/base/messages.h

```diff
@@ -148,16 +148,38 @@ namespace vcpkg::msg
 
         std::string format_examples_for_args(StringView extra_comment, const FormatArgAbi* args, std::size_t arg_count);
 
-        inline std::string get_examples_for_args(StringView extra_comment, const MessageCheckFormatArgs<>&)
+        template<::size_t M, ::size_t N>
+        inline std::string get_examples_for_args(const StringArray<M>& comment, const StringArray<N>& example)
         {
-            return extra_comment.to_string();
+            if (comment.empty())
+            {
+                return std::string(comment.begin(), comment.end());
+            }
+
+            if (example.empty())
+            {
+                return std::string(example.begin(), example.end());
+            }
+
+            const auto out = comment + StringArray(" ") + example;
+            return std::string(out.begin(), out.end());
         }
 
+        inline constexpr auto get_examples() { return StringArray{""}; }
+
         template<class Arg0, class... Args>
-        std::string get_examples_for_args(StringView extra_comment, const MessageCheckFormatArgs<Arg0, Args...>&)
+        inline constexpr auto get_examples(const Arg0& arg, Args... args)
```

Comment:
Does this damage the check that the format string is valid? (Note `MessageCheckFormatArgs` in the old code which failed to compile if the replacement strings were wrong)

Reply:
Yes, I tried to run the CMake target locally and it turned out that not all strings are valid

Reply:
@BillyONeal Do you have any suggestions on how to enable this check again?

Reply:
I think you've done what can be done already.